src/components/DateRangePicker.jsx

```diff
@@ -123,9 +136,45 @@ export default class DateRangePicker extends React.Component {
     const { onFocusChange } = this.props;
     if (!this.isOpened()) return;
 
+    this.setState({
+      isDateRangePickerInputFocused: false,
+      isDayPickerFocused: false,
+      showKeyboardShortcuts: false,
+    });
+
     onFocusChange(null);
   }
 
+  onDateRangePickerInputFocus(focusedInput) {
+    const { onFocusChange } = this.props;
+
+    if (focusedInput) {
+      this.setState({
+        isDateRangePickerInputFocused: true,
+        isDayPickerFocused: false,
+        showKeyboardShortcuts: false,
+      });
+    }
+
+    onFocusChange(focusedInput);
+  }
+
+  onDayPickerFocus() {
+    this.setState({
+      isDateRangePickerInputFocused: false,
+      isDayPickerFocused: true,
+      showKeyboardShortcuts: false,
+    });
+  }
+
+  onDayPickerBlur() {
```

Comment:
is there any potential for this to be triggered as a result of triggering the keyboard shortcut modal? or can you not trigger it from the input itself?